From 8c8b9dc444a7e087dca084bf0f17c71e4a2d5644 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Fri, 27 May 2016 10:53:09 +0800
Subject: [PATCH] ARM64: dts: rk3399: pd: enable the pd node by default

1 Remove pd_center because the ddr not allowed to power off the pd_center.
2 If the driver not used the pd. the pd will be offed after genpd init complete.
(pd disable unused)

Change-Id: I66db4df1835a48e3c0f96019bb727994e2516af9
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399.dtsi | 41 +++++++++++-------------
 1 file changed, 18 insertions(+), 23 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index 529dd2e065d2..9c14ed2784f1 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -849,35 +849,30 @@
 		reg = <0x0 0xff310000 0x0 0x1000>;
 
 		power: power-controller {
-			status = "disabled";
+			status = "okay";
 			compatible = "rockchip,rk3399-power-controller";
 			#power-domain-cells = <1>;
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			pd_center {
-				reg = <RK3399_PD_CENTER>;
-				#address-cells = <1>;
-				#size-cells = <0>;
 
-				pd_vdu {
-					reg = <RK3399_PD_VDU>;
-					pm_qos = <&qos_video_m1_r>,
-						 <&qos_video_m1_w>;
-				};
-				pd_vcodec {
-					reg = <RK3399_PD_VCODEC>;
-					pm_qos = <&qos_video_m0>;
-				};
-				pd_iep {
-					reg = <RK3399_PD_IEP>;
-					pm_qos = <&qos_iep>;
-				};
-				pd_rga {
-					reg = <RK3399_PD_RGA>;
-					pm_qos = <&qos_rga_r>,
-						 <&qos_rga_w>;
-				};
+			pd_vdu {
+				reg = <RK3399_PD_VDU>;
+				pm_qos = <&qos_video_m1_r>,
+					 <&qos_video_m1_w>;
+			};
+			pd_vcodec {
+				reg = <RK3399_PD_VCODEC>;
+				pm_qos = <&qos_video_m0>;
+			};
+			pd_iep {
+				reg = <RK3399_PD_IEP>;
+				pm_qos = <&qos_iep>;
+			};
+			pd_rga {
+				reg = <RK3399_PD_RGA>;
+				pm_qos = <&qos_rga_r>,
+					 <&qos_rga_w>;
 			};
 			pd_vio {
 				reg = <RK3399_PD_VIO>;
-- 
2.35.3

